File: rdvp/src/main/java/com/grdvp/service/interfaces/DemandeRDVServiceImpl.java
package com.grdvp.service.interfaces;

import com.grdvp.entity.DemandeRDV;
import com.grdvp.entity.Statut;
import java.util.List;

public interface DemandeRDVServiceImpl {
    void addDemand(DemandeRDV demande);
    List<DemandeRDV> searchDemand(int patientId);
    List<DemandeRDV> filterDemandByStatus(String statut);
    List<DemandeRDV> searchApointment(int patientId);
    DemandeRDV serchDemandeById(int demandeId);
    void changeDemandeStatut(DemandeRDV demande, Statut statut);
    List<DemandeRDV> getAllDemande();
}

File: src/App/service/interface/DemandeRDVServiceImpl.php
<?php

namespace App\service\interface;

use App\entity\DemandeRDV;
use App\entity\Patient;
use App\entity\Statut;

interface DemandeRDVServiceImpl
{
    public function addDemand(DemandeRDV $demande): void;

    public function searchDemand(Patient $patient): array;
    
    public function filterDemandByStatus(Patient $patient, string $statut): array;

    public function searchApointment(Patient $patient): array;

    public function searchDemandeById(int $id): DemandeRDV;

    public function getAllDemandes(): array;
    
    public function changedeamndeStatus(DemandeRDV $demande, Statut $statut): void;
}